### PR TITLE
Tidy spacing guidance, and document how to do negative spacing.

### LINKED ISSUE #1896: Document how to use govuk-spacing for negative spacing
Document the way that the `govuk-spacing` function can now be used to output negative spacing introduced in https://github.com/alphagov/govuk-frontend/pull/2348.

### PR DESCRIPTION
**Pull request by 36degrees**
Closes #1896

### CODE REVIEW

File: src/styles/spacing/index.md.njk [hunk 1]
```diff
@@ -106,20 +106,42 @@ The last part of the class represents the value you want to apply. For example,
 
 If you’re building your own components and want to reference the spacing scale directly in your SCSS file you can use the spacing scale through a mixin or a function.
 
+### Using the responsive spacing mixin
+
+To use the responsive spacing scale, include the [`govuk-responsive-margin`](https://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-responsive-margin) or [`govuk-responsive-padding`](https://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-responsive-padding) mixins.
+
+For example, to apply 30px padding on large screens, and 20px on small screens, use:
```
Comment (calvin-lau-sig7) [inline src/styles/spacing/index.md.njk:113]:
```suggestion
For example, to apply spacing unit 6 for 30px padding on large screens, and 20px on small screens, use:
```

File: src/styles/spacing/index.md.njk [hunk 2]
```diff
@@ -106,20 +106,42 @@ The last part of the class represents the value you want to apply. For example,
...
+
+```scss
+@include govuk-responsive-padding(6);
+```
+
+To apply margin or padding in a single direction, pass the direction as the second argument.
+
+For example, to apply a 30px bottom margin on large screens and a 20px bottom margin on small screens, use:
```
Comment (calvin-lau-sig7) [inline src/styles/spacing/index.md.njk:121]:
```suggestion
You can also add an argument to apply margin or padding in a single direction.

For example, to apply spacing unit 6 for a 30px bottom margin on large screens and a 20px bottom margin on small screens, use:
```

File: src/styles/spacing/index.md.njk [hunk 3]
```diff
@@ -106,20 +106,42 @@ The last part of the class represents the value you want to apply. For example,
...
+@include govuk-responsive-margin(6, "bottom");
+```
+
 ### Using the static spacing function
 
-If you want to use static values that will not change based on breakpoints you can reference the static spacing scale function like this:
+If you want to apply 'large screen' spacing at all breakpoints, use the [`govuk-spacing` function](https://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-spacing) instead.
 
-`padding-top: govuk-spacing(6)`
+For example, to apply 30px top padding on both large and small screens, use:
```
Comment (calvin-lau-sig7) [inline src/styles/spacing/index.md.njk:131]:
```suggestion
For example, to apply spacing unit 6 for 30px top padding on both large and small screens, use:
```

File: src/styles/spacing/index.md.njk [hunk 4]
```diff
@@ -106,20 +106,42 @@ The last part of the class represents the value you want to apply. For example,
...
 
+```scss
+padding-top: govuk-spacing(6);
+```
 
-### Using the responsive spacing mixin
+For negative spacing, you can also call it with a negated spacing unit.
```
Comment (calvin-lau-sig7) [inline src/styles/spacing/index.md.njk:137]:
```suggestion
For negative spacing, use a negative spacing unit number.
```

File: src/styles/spacing/index.md.njk [hunk 5]
```diff
@@ -106,20 +106,42 @@ The last part of the class represents the value you want to apply. For example,
...
 
-If you want to use the responsive spacing scale you can reference the mixin.
+For example, to apply a negative 15px top margin on both large and small screens, use:
```
Comment (calvin-lau-sig7) [inline src/styles/spacing/index.md.njk:139]:
```suggestion
For example, to apply spacing unit -3 for a negative 15px top margin on both large and small screens, use:
```

File: src/styles/spacing/index.md.njk [hunk 6]
```diff
@@ -106,20 +106,42 @@ The last part of the class represents the value you want to apply. For example,
...
+For example, to apply spacing unit 6 for a 30px bottom margin on large screens and a 20px bottom margin on small screens, use:
+
+```scss
+@include govuk-responsive-margin(6, "bottom");
+```
+
 ### Using the static spacing function
 
-If you want to use static values that will not change based on breakpoints you can reference the static spacing scale function like this:
+If you want to apply 'large screen' spacing at all breakpoints, use the [`govuk-spacing` function](https://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-spacing) instead.
```
Comment (lfdebrux) [inline src/styles/spacing/index.md.njk:129]: I don't think the quote marks are needed here, given that the term large screen is defined earlier on in the page.

**Review by lfdebrux — APPROVED** [review]
Looks good to me.

File: src/styles/spacing/index.md.njk (hunk 6 above)
Comment (calvin-lau-sig7) [reply src/styles/spacing/index.md.njk:129]: Good catch, thanks @lfdebrux!

```suggestion
If you want to apply large screen spacing at all breakpoints, use the [`govuk-spacing` function](https://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-spacing) instead.
```

**Pull request merged**